utf16fix.h

```diff
@@ -1,6 +1,12 @@
 #include <stdbool.h>
-#include <uchar.h>
+#include <stddef.h>
+#include <stdint.h>
 
+#if defined(__APPLE__)
+typedef uint16_t char16_t;
+#else
+#include <uchar.h>
+#endif
```

Comment:
You can remove the `uchar.h` includes from the other files and just have the platform-specific code at this one point.